client/app/scripts/reducers/root.js

```diff
@@ -688,7 +688,7 @@ export function rootReducer(state = initialState, action) {
         pinnedMetricType: action.state.pinnedMetricType,
       });
       if (action.state.topologyOptions) {
-        state = state.set('topologyOptions', fromJS(action.state.topologyOptions));
+        state = state.mergeDeep(fromJS({ topologyOptions: action.state.topologyOptions }));
```

Comment:
As @foot and I discussed, it would be safer to also merge `action.state.topologyOptions` with `getDefaultTopologyOptions(state)` in here to make sure everything works fine if we switch from non-default to default topology options (otherwise `action.state.topologyOptions` would be empty and it wouldn't overwrite the current state).

p.s. Turns out the real issue here was not taking the default options into account in the first place, once they are merged with the URL state, they provide a complete description of topology options state, so it shouldn't matter whether we use `set` or `mergeDeep` :)

Reply:
Yeah, turns out that action is only fired onScopeLoad rather than any subsequent ui actions, which is sad. But I will update as discussed anyway in case that changes in the future! 👍